### PR TITLE
Adds L85A2 and Butt launcher

### PR DESCRIPTION
**Pull request by lcass**
Adds two new weapons to the game.
L85A2: A constructable gun that fires with fairly low stats and has a variant usable by security , the stamina rifle. With lethal ammunition it does 15 damage per shot with 10% embed chance , with stamina ammunition it does 3 damage per shot and between 5 and 20 stamina loss per hit. The rifle has two firing modes 2 shot and semi auto. Requires plasteel , screwdriver , weldingtool , modular reciever and metal rods in order to construct. The initial frame is produced for 25000 metal in a hacked autolathe. Lethal ammunition can only be obtained from an emagged or hacked sec tech vendor.

Butt launcher: Coincides with the buttdate as it adds a new gimmick weapon that fires butts at people. Constructed through a hacked autolathe and a mouse trap , holds 1 butt at a time and has a 5% chance to smack someone in the face with the butt stunning them for a short period. When butts are thrown they now fart on impact.

Other info: 2 L85A2's are located in the ballistic gun closet , both loaded with stamina rounds.

### CODE REVIEW

**Comment by Chronitonity** [comment]
> L85A2: A constructable gun that fires with fairly low stats and has a variant usable by security , the stamina rifle

Words cannot describe how :-1: I am about this. Why would add a build-able rifle that does 15 damage per shot. Why would you give security a stamina rifle when they already have an assload of guns.

> Other info: 2 L85A2's are located in the ballistic gun closet , both loaded with stamina rounds.
> W h y.

God no

**Comment by lcass** [comment]
Thanks to oprayx for the sprites.

**Comment by lcass** [comment]
Considering in order to get lethal ammo you would need to get into security you might aswell steal an egun which does more damage. The only ammo easily obtainable are the stamina rounds.

**Comment by Chronitonity** [comment]
Which is still insanely powerful and I'd rather we didn't add craftable ballistic semi autorifles and give some to security and let others be crafted.

**Comment by lcass** [comment]
They aren't exactly easy to craft , they required a modular reciever which is research requiring tech. By the point you have that you basically have access to xray guns etc.

**Comment by Chronitonity** [comment]
Still :-1: people are going to steal the ones from sec and load up on lethal weapons by hacking the vendors. Theres no upside to adding these.

**Comment by lcass** [comment]
That is more difficult that getting an egun from the armoury or even ordering one from cargo , they do less damage than eguns more difficult to find ammo for and there are less of them.

**Comment by Chronitonity** [comment]
Seeing how you could store a ton of ammo in your backpack and never run out, i think its a decent enough reason to take them. We dont need more guns.

**Comment by Pyko1** [comment]
Now I just need to snatch that butt launcher code and turn it in to a pneumatic potato (sized object) cannon.

**Comment by Weisyl** [comment]
In the current state right now, I can't see this being added. If we made it able to fire non-lethal rubber bullets only this might have a better chance of being added, but still. The code on this is also pretty strange compared to other guns.

**Comment by lcass** [comment]
Code has been adjusted to be less ?strange?

**Comment by ghost** [comment]
i sprited teh guns, please add. honestly with the removal of the pistols from spess these could fill a void in the armoury

**Comment by lcass** [comment]
Lowered the total number of lethal mags on the map to 8 .

**Comment by SplergeBucket** [comment]
Butt Launcher might be fine but the Security Gun sounds like an unnecessary shitty addition.

**Comment by danhibiki** [comment]
I wouldn't mind a ballistic non-lethal weapon, not sure how I feel about having lethal ammo readily available right now though.

**Comment by lcass** [comment]
I can remove the lethal ammo and make it only available to traitors or that it would require fairly high research tech beyond the point of xray weapons.

**Comment by Pyko1** [comment]
I demand that the launched butts have (really small) embed chance.

**Comment by Chronitonity** [comment]
All butts have a 5% chance to embed Pykey

**Comment by Pyko1** [comment]
Yes! I can finally be a butthead like chron!

**Comment by Nebyab** [comment]
Whoops, totally forgot to get back to this. The butt-launcher I'm 100% infavor of, but I do have some qualms about the new gun. Sec has plenty of non-lethal options available, they don't need more. I'm not against the idea of adding a craftable gun, but it needs to be reworked before I could okay its implementation.

File: code/modules/projectiles/ammunition/magazines.dm [hunk 1]
```diff
@@ -286,4 +286,26 @@ obj/item/ammo_box/magazine/tommygunm45
...
+	ammo_type = /obj/item/ammo_casing/shotgun/stunslug
+//l85 ammunition
+/obj/item/ammo_box/magazine/l85
+	name = "L85A2 5.56x45mm magazine"
+	icon = 'icons/obj/buttlauncher.dmi'
+	icon_state = "full"
+	origin_tech = "combat=3"
+	ammo_type = /obj/item/ammo_casing/a556
+	caliber = "556"
+	max_ammo = 30
```
Comment (SplergeBucket) [inline code/modules/projectiles/ammunition/magazines.dm:298]: 30 ammo, 30 ammo for a craftable gun which will easily be obtainable by science and no one else.

File: code/modules/projectiles/guns/projectile/automatic.dm [hunk 2]
```diff
@@ -308,4 +308,79 @@
 	mag_type = /obj/item/ammo_box/magazine/tommygunm45
 	fire_sound = 'sound/weapons/Gunshot_smg.ogg'
 	can_suppress = 0
-	burst_size = 4
\ No newline at end of file
+	burst_size = 4
+
+/obj/item/weapon/gun/projectile/automatic/l85a2
```
Comment (SplergeBucket) [inline code/modules/projectiles/guns/projectile/automatic.dm:313]: We honestly don't need more buffed weapons like this; certainly not ones which only science can produce. This is a dumb unbalanced addition.

File: code/modules/projectiles/guns/projectile/launchers.dm [hunk 3]
```diff
@@ -37,4 +37,108 @@
...
-	return
\ No newline at end of file
+	return
+/obj/item/weapon/gun/buttlauncher
+	name = "Butt Launcher"
+	desc = "For HONKing on an extreme level!"
+	icon = 'icons/obj/buttlauncher.dmi'
+	icon_state = "buttempty"
+	item_state = "riotgun"
+	w_class = 2.0
```
Comment (SplergeBucket) [inline code/modules/projectiles/guns/projectile/launchers.dm:47]: This is not a proper weight class for this item. It should be at the very least 3 or 4. Whichever one doesn't allow it in the backpack due to bulky item state is preferable.

**Comment by SplergeBucket** [comment]
Overall I like the butt launcher, I think that's pretty solid.
I'm not interested in the rifle at all.

**Comment by lcass** [comment]
Made the gun not fit into a backpack , only non lethal rounds for non traitors.

**Comment by lcass** [comment]
considering we are rebasing It wouldn't hurt to try it. Up to the heads though.

**Comment by lcass** [comment]
Well getting no responses after updates are made and it generally being ignored doesn't help. This is dead along with other stuff welp.

**Pull request closed without merging**